sdk/ml/azure-ai-ml/azure/ai/ml/_schema/component/data_transfer_component.py

```diff
@@ -0,0 +1,223 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+# pylint: disable=unused-argument,no-self-use,protected-access,no-member
+
+from copy import deepcopy
+
+import yaml
+from marshmallow import INCLUDE, fields, post_load, validates, ValidationError
+
+from azure.ai.ml._schema.assets.asset import AnonymousAssetSchema
+from azure.ai.ml._schema.component.component import ComponentSchema
+from azure.ai.ml._schema.component.input_output import InputPortSchema, OutputPortSchema
+from azure.ai.ml._schema.core.schema_meta import PatchedSchemaMeta
+from azure.ai.ml._schema.core.fields import FileRefField, StringTransformedEnum, NestedField
+from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY, AssetTypes
+from azure.ai.ml.constants._component import ComponentSource, NodeType, DataTransferTaskType, DataCopyMode, ExternalDataType
+
+
+class DataTransferComponentSchemaMixin(ComponentSchema):
+    type = StringTransformedEnum(allowed_values=[NodeType.DATATRANSFER])
+
+
+class DataTransferCopyComponentSchema(DataTransferComponentSchemaMixin):
+    task = StringTransformedEnum(allowed_values=[DataTransferTaskType.COPY_DATA], required=True)
+    data_copy_mode = StringTransformedEnum(allowed_values=[DataCopyMode.MERGE_WITH_OVERWRITE,
+                                                           DataCopyMode.FAIL_IF_CONFLICT], required=True)
+    inputs = fields.Dict(
+        keys=fields.Str(),
+        values=NestedField(InputPortSchema),
+    )
+
+
+class SinkSourceSchema(metaclass=PatchedSchemaMeta):
+    type = StringTransformedEnum(allowed_values=[ExternalDataType.FILE_SYSTEM, ExternalDataType.DATABASE],
+                                 required=True)
+
+
+class InputsOutputsSchema(metaclass=PatchedSchemaMeta):
```

Comment:
only Import/Export has this type limit?  May reflect that in classname or add docstring